course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,192 @@
+#include "my_string.h"
+
+#include <cstring>
+#include <memory>
+
+MyString::MyString() 
+	:
+	mBuffer(nullptr)
+{
+}
+
+MyString::MyString(const MyString& that)
+{
+	if (that.empty())
+	{
+		mBuffer = nullptr;
+	}
+	else
+	{
+		mBuffer = std::make_unique<char[]>(std::strlen(that.mBuffer.get()) + 1);
+		std::strcpy(mBuffer.get(), that.mBuffer.get());
+	}
+}
+
+MyString::MyString(MyString&& that)
+{
+	mBuffer.swap(that.mBuffer);
+}
+
+MyString::MyString(const char* str)
+{
+	if (str == nullptr || str[0] == '\0')
+	{
+		mBuffer = nullptr;
+	}
+	else
+	{
+		mBuffer = std::make_unique<char[]>(std::strlen(str) + 1);
+		std::strcpy(mBuffer.get(), str);
+	}
+}
+
+MyString& MyString::operator=(const MyString& that)
+{
+	if (this == &that)
```

Comment:
Think of the "don't pay for what you are not using" principle. How frequently would this mistake happen (very rarely) vs how often do you need to pay the price of the check (always)

I think it would be nice if your function simply worked correctly in such a case, but it's fine if it doesn't